src/api/router/RouterHistory.ts

```diff
@@ -0,0 +1,28 @@
+import { ListLikeObvserver } from "../../internals/ListLikeObserver";
+
+export type HistoryItem = { route: string, payload?: any };
+
+export class RouterHistory<T extends HistoryItem = HistoryItem> extends ListLikeObvserver<T> {
+
+  public push(item: T) {
+    const source = Array.from(this.source);
```

Comment:
This would replace the source even if it's a list. I'd rather deal with this using instanceof